File: Editorial/496_Next_Greater_Element_I/CPP/2_Better_Brute_Force.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> hash;
        for (int i = 0; i < nums2.size(); i++) {
            hash[nums2[i]] = i;
        }

        vector<int> res(nums1.size());
        int j;
        for (int i = 0; i < nums1.size(); i++) {
            for (j = hash[nums1[i]] + 1; j < nums2.size(); j++) {
                if (nums1[i] < nums2[j]) {
                    res[i] = nums2[j];
                    break;
                }
            }
            if (j == nums2.size()) {
                res[i] = -1;
            }
        }

        return res;
    }
};

File: Editorial/496_Next_Greater_Element_I/Python3/2_Better_Brute_Force.py
class Solution:
    def nextGreaterElement(
        self, nums1: List[int], nums2: List[int]
    ) -> List[int]:
        hash_table = {num: i for i, num in enumerate(nums2)}

        res = [0] * len(nums1)
        for i, num in enumerate(nums1):
            j = hash_table[num] + 1
            while j < len(nums2):
                if num < nums2[j]:
                    res[i] = nums2[j]
                    break
                j += 1
            else:
                res[i] = -1

        return res
